src/neptune/api/operations.py

```diff
@@ -0,0 +1,200 @@
+#
+# Copyright (c) 2024, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+__all__ = [
+    "ProtoSerializable",
+    "Run",
+    "FloatValue",
+    "LogFloats",
+    "AssignInteger",
+    "AssignFloat",
+    "AssignBool",
+    "AssignString",
+    "AssignDatetime",
+    "RunOperation",
+]
+
+
+import abc
+from dataclasses import dataclass
+from datetime import datetime
+from typing import (
+    List,
+    Optional,
+)
+
+from google.protobuf import timestamp_pb2
+
+import neptune.api.proto.neptune_pb.ingest.v1.common_pb2 as common_pb2
+import neptune.api.proto.neptune_pb.ingest.v1.pub.ingest_pb2 as ingest_pb2
+
+
+class ProtoSerializable(abc.ABC):
```

Comment:
We have mixed two layers. This is an abstract class for API operation and should be named without any mention of `proto`.

How about consumer/modifier method `to_proto(self, run_operation: RunOperation) -> RunOperation? instead of `create` and `update`.